LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Switch statements and expressions with a sealed class as a
+/// matched type are always exhaustive if the set of cases is exhaustive
+///
+/// @description Check that it is no compile-time error if the matched value
+/// type of a switch statement is an extension type with a sealed class as a
+/// representation type and the set of cases is an exhaustive set of variable
+/// patterns
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+sealed class A {}
+
+class B extends A {}
+
+class C extends B {}
+
+extension type AET1(A _) {}
+extension type AET2(A _) implements A {}
+extension type BET1(B _) {}
+extension type BET2(B _) implements B {}
+
+String test1(AET1 a) {
+  switch (a) {
+    case B _: return 'B';
+  }
+}
```

Comment:
This one is blocked on https://github.com/dart-lang/language/issues/3534: It hasn't yet been decided whether we should use the extension type erasures (as I'd prefer) during flow analysis in order to determine whether or not it is possible to reach the end of the function body in this example.

So we'll have to put this one in a waiting position until https://github.com/dart-lang/language/issues/3534 has been settled.

Reply:
https://github.com/dart-lang/language/issues/3534 has been settled. Now this test should work

Reply:
Right!